src/OMSimulatorLib/SystemSC.cpp

```diff
@@ -625,7 +625,11 @@ oms_status_enu_t oms::SystemSC::stepUntil(double stopTime, void (*cb)(const char
     for (int i=0; i < fmus.size(); ++i)
     {
       fmistatus = fmi2_import_completed_integrator_step(fmus[i]->getFMU(), fmi2_true, &callEventUpdate[i], &terminateSimulation[i]);
-      if (fmi2_status_ok != fmistatus) logError_FMUCall("fmi2_import_completed_integrator_step", fmus[i]);
+      if (fmi2_status_ok != fmistatus)
+      {
+        logError_FMUCall("fmi2_import_completed_integrator_step", fmus[i]);
+        return oms_status_error;
+      }
```

Comment:
You could just write `return logError_FMUCall ...` because log_Error returns the error status to simplify those code blocks.

Reply:
Do we actually release all resources correctly if exiting `stepUntil` in this way?

Reply:
I bet we don't.

Reply:
Is it something we should be worried about?